hail/python/test/hailtop/batch/test_batch.py

```diff
@@ -88,6 +88,25 @@ def test_single_job(self):
 
             assert self.read(output_file.name) == msg
 
+    def test_single_job_with_shell(self):
+        with tempfile.NamedTemporaryFile('w') as output_file:
+            msg = 'hello world'
+
+            b = self.batch()
+            j = b.new_job(shell='/bin/sh')
+            j.command(f'echo "{msg}" > {j.ofile}')
+
+            b.write_output(j.ofile, output_file.name)
+            b.run()
+
+            assert self.read(output_file.name) == msg
+
+    def test_single_job_with_nonsense_shell(self):
+        b = self.batch()
+        j = b.new_job(shell='bin/ajdsfoijasidojf')
+        j.command(f'echo "hello"')
+        self.assertRaises(Exception, b.run())
```

Comment:
Good call. It takes a callable, should not have been invoked here. Can't use .status() here because this is a LocalBackend test and sp.check_call raises. It would be nice to have similar .status() available on LocalBackend.